app/models.py

```diff
@@ -2551,3 +2551,90 @@ def serialize(self):
             "original_filename": self.original_filename,
             "page_count": self.page_count,
         }
+
+
+class UnsubscribeRequestReport(db.Model):
+    __tablename__ = "unsubscribe_request_report"
+    """
+    CREATE VIEW unsubscribe_request_report AS
+    SELECT
+        service_id,
+        unsubscribe_request_report_id as id,
+        min(created_at) as earliest_timestamp,
+        max(created_at) as latest_timestamp,
+        (case when downloaded_at is null then false else true end) as has_been_downloaded
+        count(*)
+    FROM unsubscribe_requests
+    group by service_id, unsubscribe_request_report_id
+    """
+
+    # Tell alembic not to create this as a table. We have a migration where we manually set this up as a view.
+    # This is custom logic we apply - not built-in logic. See `migrations/env.py`
+    __table_args__ = {"info": {"managed_by_alembic": False}}
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True)
+    service_id = db.Column(UUID(as_uuid=True), db.ForeignKey("services.id"), nullable=False)
+    service = db.relationship(Service, backref=db.backref("unsubscribe_requests"))
+
+    earliest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    latest_timestamp = db.Column(db.DateTime, nullable=False, default=datetime.datetime.utcnow)
+    has_been_downloaded = db.Column(db.Boolean, nullable=False)
+    count = db.Column(db.Integer, nullable=False)
+
```

Comment:
We would need a boolean `processed` attribute and a `processed_at` . On slide 4 of the proposed user interface design, a key part of the journey is the user marking that the report has been actioned and that the contained unsubscribe requests have been actioned.

Reply:
changed `downloaded_at` and `has_been_downloaded` to `processed_at` and `has_been_processed` to help resolve this.

but i'm not sure if `processed_at` is too generic a term. Who has processed it? What does processing mean?

I guess processing itself and what that means is up to the service user - but maybe your `processed_by_service` idea has legs 🦵 🦵

Reply:
Maybe it could be `actioned_at` and `has_been_actioned` ?

Reply:
i think i like gozie's idea - `processed_by_service_at`. gonna go with that for now